addons/website_slides/controllers/slide.py

```diff
@@ -0,0 +1,82 @@
+                'nb_attempts': slide_user.quiz_attempts_count if slide_user else 0,
+                'possible_rewards': [slide.quiz_first_attempt_reward, slide.quiz_second_attempt_reward, slide.quiz_third_attempt_reward, slide.quiz_fourth_attempt_reward],
+                'reward': (slide[possible_points[slide_user.quiz_attempts_count]] if slide_user.quiz_attempts_count < 3 else slide.quiz_fourth_attempt_reward) if slide_user else slide.quiz_first_attempt_reward
+            }
+            return res
+
+    @http.route('/slide/quiz/submit', type="json", auth="user", website=True)
+    def submit_quiz(self, slide_id, answer_ids,**kw):
+        slide = request.env['slide.slide'].browse(slide_id)
+        answers = request.env['slide.answer'].search([('id', 'in', answer_ids)])
+        good_answers = answers.search([('id', 'in', answer_ids),('is_correct', '=', True)])
+        bad_answers = answers - good_answers
+        slide_partner = request.env['slide.slide.partner'].search([('slide_id', '=', slide_id),('partner_id', '=', request.env.user.partner_id.id)])
+        possible_points = ["quiz_first_attempt_reward","quiz_second_attempt_reward","quiz_third_attempt_reward"]
+        points = 0
+        if not slide_partner:
+            slide_partner = request.env['slide.slide.partner'].sudo().create({
+                "slide_id": slide.id,
+                "partner_id": request.env.user.partner_id.id,
+                "channel_id": slide.channel_id.id,
+                "completed": False
+            })
+        if not slide_partner.completed:
+            if slide_partner.quiz_attempts_count < 3:
+                points = slide[possible_points[slide_partner.quiz_attempts_count]]
+            else:
+                points = slide.quiz_fourth_attempt_reward
+            slide_partner.sudo().write({
+                'quiz_attempts_count': slide_partner.quiz_attempts_count if not bad_answers else slide_partner.quiz_attempts_count + 1,
+                'points_won': points if not bad_answers else 0,
+                'completed':  not bad_answers
+            })
+            return {
+                'goodAnswers':[good_answer.id for good_answer in good_answers],
+                'badAnswers': [bad_answer.id for bad_answer in bad_answers],
+                'passed': not bad_answers,
+                'points': points if not bad_answers else 0,
+                'attempts_count': slide_partner.quiz_attempts_count if slide_partner else 0
+                }
+        return {
+            'error': "You already passed this quiz"
+        }
+
+    @http.route('/slide/completed/<int:slide_id>', website=True, type="http", auth="user")
+    def mark_as_completed(self, slide_id, next_slide=None, **kw):
+        slide = request.env['slide.slide'].browse(slide_id)
+        slide.action_set_completed()
```

Comment:
Can't you call the public method on the record instead of creating a controller method for it?